Automattic/jetpack
# Search: Add search product tier pricing

Fixes #23234 

#### Changes proposed in this Pull Request:
Adds an API for site tier pricing, which exposes functionality added in #23892. <strike>The PR is based on #23892 and #23809</strike> - both merged.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Navigate to a site with only Search plugin (no Jetpack plugin)
- Open `/wp-admin/admin.php?page=jetpack-search`
- Ensure the pricing correctly reflects the tier
- add `return 101;` as first line of `Stats::estimate_count`
- Ensure the pricing correctly reflects the tier
![image](https://user-images.githubusercontent.com/1425433/162868766-99cbc4cb-185b-41fd-85d6-5a08cbcd7402.png)
